File: msglnk-web/src/main/java/msglnk/data/entity/BaseEntity.java
package msglnk.data.entity;

import org.codehaus.groovy.runtime.DefaultGroovyMethods;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
public class BaseEntity {

    @Id
    @GeneratedValue
    private Long uid;

    public Long getUid() {
        return uid;
    }

    public void setUid(Long uid) {
        this.uid = uid;
    }

    public boolean equals(Object o) {
        if (DefaultGroovyMethods.is(this, o)) return true;
        if (!getClass().equals(o.getClass())) return false;

        BaseEntity that = (BaseEntity) o;

        if (uid != that.uid) return false;

        return true;
    }

    public int hashCode() {
        return (uid != null ? uid.hashCode() : 0);
    }
}

File: src/main/scala/msglnk/data/BaseEntity.scala
/**
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package msglnk.data

import javax.persistence.{Id, GeneratedValue, MappedSuperclass}
import java.lang.Long
import scala.reflect.BeanProperty

@MappedSuperclass
class BaseEntity {
    @Id
    @GeneratedValue
    @BeanProperty
    var uid: Long = _

    override def equals(that: Any): Boolean = {
        that.isInstanceOf[BaseEntity] && this.hashCode() == that.asInstanceOf[BaseEntity].hashCode()
    }

    override def hashCode: Int = {
        if (uid == null) 0 else uid.hashCode
    }
}
